polkadot/node/network/statement-distribution/src/v2/tests/requests.rs

```diff
@@ -1103,6 +1103,560 @@ fn peer_reported_for_providing_statements_with_wrong_validator_id() {
 	});
 }
 
+#[test]
+fn disabled_validators_added_to_unwanted_mask() {
+	let group_size = 3;
+	let config = TestConfig {
+		validator_count: 20,
+		group_size,
+		local_validator: LocalRole::Validator,
+		async_backing_params: None,
+	};
+
+	let relay_parent = Hash::repeat_byte(1);
+	let peer_disabled = PeerId::random();
+	let peer_b = PeerId::random();
+
+	test_harness(config, |state, mut overseer| async move {
+		let local_validator = state.local.clone().unwrap();
+		let local_group_index = local_validator.group_index.unwrap();
+		let local_para = ParaId::from(local_group_index.0);
+		let other_group_validators = state.group_validators(local_group_index, true);
```

Comment:
Why are we passing `local_group_index` here? Shouldn't the grid peer be in a different group?

Reply:
The we don't grid-specific disabling logic, so here the disabled peer is cluster peer.